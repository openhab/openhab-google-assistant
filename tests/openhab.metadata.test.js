
const OpenHAB = require('../functions/openhab.js').OpenHAB;

describe('Test SYNC with Metadata', () => {
  test('Light Devices', async () => {
    const items = [
      {
        "state": "OFF",
        "type": "Switch",
        "name": "MySwitch",
        "label": "SwitchLight",
        "metadata": {
          "ga": {
            "value": "Light",
            "config": {
              "name": "Light Switch"
            }
          },
          "synonyms": {
            "value": "Testlight,Cool Light"
          }
        }
      },
      {
        "state": "0",
        "type": "Dimmer",
        "name": "MyDimmer",
        "label": "DimmLight",
        "metadata": {
          "ga": {
            "value": "Light"
          }
        }
      },
      {
        "state": "0,0,0",
        "type": "Color",
        "name": "MyLight",
        "label": "ColorLight",
        "metadata": {
          "ga": {
            "value": "Light"
          }
        }
      },
      {
        "members": [],
        "state": "NULL",
        "type": "Group",
        "groupType": "Switch",
        "name": "MyLightGroup",
        "label": "GroupLight",
        "metadata": {
          "ga": {
            "value": "Light"
          }
        }
      },
      {
        "members": [],
        "state": "NULL",
        "type": "Group",
        "groupType": "Dimmer",
        "name": "MyDimmerGroup",
        "label": "GroupDimmer",
        "metadata": {
          "ga": {
            "value": "Light"
          }
        }
      },
      {
        "members": [],
        "state": "NULL",
        "type": "Group",
        "groupType": "Color",
        "name": "MyColorGroup",
        "label": "GroupColor",
        "metadata": {
          "ga": {
            "value": "Light"
          }
        }
      }
    ];

    const getItemsMock = jest.fn();
    getItemsMock.mockReturnValue(Promise.resolve(items));

    const apiHandler = {
      getItems: getItemsMock
    };

    const payload = await new OpenHAB(apiHandler).handleSync();

    expect(getItemsMock).toHaveBeenCalledTimes(1);
    expect(payload).toMatchSnapshot();
  });

  test('Fan Device', async () => {
    const items = [
      {
        "state": "50",
        "metadata": {
          "ga": {
            "value": "FAN",
            "config": {
              "ordered": true,
              "speeds": "0\u003dnull:off,50\u003dslow,100\u003dfull:fast",
              "lang": "en"
            }
          }
        },
        "type": "Dimmer",
        "name": "MyFan",
        "label": "My Fan",
        "tags": []
      }
    ];
    const getItemsMock = jest.fn();
    getItemsMock.mockReturnValue(Promise.resolve(items));

    const apiHandler = {
      getItems: getItemsMock
    };

    const payload = await new OpenHAB(apiHandler).handleSync();

    expect(getItemsMock).toHaveBeenCalledTimes(1);
    expect(payload).toMatchSnapshot();
  });

  test('Scene Device', async () => {
    const items = [
      {
        "state": "OFF",
        "metadata": {
          "ga": {
            "value": "Scene"
          }
        },
        "type": "Switch",
        "name": "MyScene",
        "label": "My Scene",
        "tags": []
      }
    ];
    const getItemsMock = jest.fn();
    getItemsMock.mockReturnValue(Promise.resolve(items));

    const apiHandler = {
      getItems: getItemsMock
    };

    const payload = await new OpenHAB(apiHandler).handleSync();

    expect(getItemsMock).toHaveBeenCalledTimes(1);
    expect(payload).toMatchSnapshot();
  });
});

describe('Test QUERY with Metadata', () => {
  test('Single Light Device', async () => {
    const item =
    {
      "state": "OFF",
      "type": "Switch",
      "name": "MySwitch",
      "metadata": {
        "ga": {
          "value": "Light"
        }
      }
    };

    const getItemMock = jest.fn();
    getItemMock.mockReturnValue(Promise.resolve(item));

    const apiHandler = {
      getItem: getItemMock
    };

    const payload = await new OpenHAB(apiHandler).handleQuery([{
      "id": "MySwitch"
    }]);

    expect(getItemMock).toHaveBeenCalledTimes(1);
    expect(payload).toStrictEqual({
      "devices": {
        "MySwitch": {
          "on": false,
          "online": true,
        },
      },
    });
  });

  test('Multiple Light Devices', async () => {
    const item1 =
    {
      "state": "OFF",
      "type": "Switch",
      "name": "MySwitch",
      "metadata": {
        "ga": {
          "value": "Light"
        }
      }
    };

    const item2 =
    {
      "state": "20",
      "type": "Dimmer",
      "name": "MyDimmer",
      "metadata": {
        "ga": {
          "value": "Light"
        }
      }
    };

    const getItemMock = jest.fn();
    getItemMock.mockReturnValueOnce(Promise.resolve(item1))
      .mockReturnValueOnce(Promise.resolve(item2));

    const apiHandler = {
      getItem: getItemMock
    };

    const payload = await new OpenHAB(apiHandler).handleQuery([{
      "id": "MySwitch"
    }, {
      "id": "MyDimmer"
    }]);

    expect(getItemMock).toHaveBeenCalledTimes(2);
    expect(payload).toStrictEqual({
      "devices": {
        "MySwitch": {
          "on": false,
          "online": true,
        },
        "MyDimmer": {
          "on": true,
          "brightness": 20,
          "online": true,
        },
      },
    });
  });

  test('Fan Device', async () => {
    const item = {
      "state": "50",
      "metadata": {
        "ga": {
          "value": "FAN",
          "config": {
            "ordered": true,
            "speeds": "0\u003dnull:off,50\u003dslow,100\u003dfull:fast",
            "lang": "en"
          }
        }
      },
      "type": "Dimmer",
      "name": "MyFan",
    };
    const getItemMock = jest.fn();
    getItemMock.mockReturnValue(Promise.resolve(item));

    const apiHandler = {
      getItem: getItemMock
    };

    const payload = await new OpenHAB(apiHandler).handleQuery([{
      "id": "MyFan"
    }]);

    expect(getItemMock).toHaveBeenCalledTimes(1);
    expect(payload).toStrictEqual({
      "devices": {
        "MyFan": {
          "on": true,
          "currentFanSpeedSetting": "50",
          "online": true,
        },
      },
    });
  });

  test('Thermostat Device', async () => {
    const item =
    {
      "state": "NULL",
      "type": "Group",
      "name": "MyThermostat",
      "label": "Thermostat",
      "metadata": {
        "ga": {
          "value": "Thermostat",
          "config": {
            "useFahrenheit": true,
            "modes": "off=OFF:WINDOW_OPEN,heat=COMFORT:BOOST,eco=ECO,on=ON,auto=AUTOMATIC"
          }
        }
      },
      "members": [{
        type: 'Number',
        metadata: {
          ga: {
            value: 'thermostatTemperatureAmbient'
          }
        },
        state: '10'
      }, {
        name: 'MyTargetTemperature',
        type: 'Number',
        metadata: {
          ga: {
            value: 'thermostatTemperatureSetpoint'
          }
        },
        state: '10'
      }, {
        type: 'Number',
        metadata: {
          ga: {
            value: 'thermostatMode'
          }
        },
        state: 'COMFORT'
      }, {
        type: 'Number',
        metadata: {
          ga: {
            value: 'thermostatHumidityAmbient'
          }
        },
        state: '50'
      }]
    };
    const getItemMock = jest.fn();
    getItemMock.mockReturnValue(Promise.resolve(item));

    const apiHandler = {
      getItem: getItemMock
    };

    const payload = await new OpenHAB(apiHandler).handleQuery([{
      "id": "MyThermostat"
    }]);

    expect(getItemMock).toHaveBeenCalledTimes(1);
    expect(payload).toStrictEqual({
      "devices": {
        "MyThermostat": {
          "thermostatHumidityAmbient": 50,
          "thermostatMode": "heat",
          "thermostatTemperatureAmbient": -12.2,
          "thermostatTemperatureSetpoint": -12.2,
          "online": true
        },
      },
    });
  });
});

describe('Test EXECUTE with Metadata', () => {
  test('ThermostatTemperatureSetpoint', async () => {
    const item =
    {
      "state": "NULL",
      "type": "Group",
      "name": "MyThermostat",
      "label": "Thermostat",
      "metadata": {
        "ga": {
          "value": "Thermostat",
          "config": {
            "useFahrenheit": true
          }
        }
      },
      "members": [{
        type: 'Number',
        metadata: {
          ga: {
            value: 'thermostatTemperatureAmbient'
          }
        },
        state: '10'
      }, {
        name: 'MyTargetTemperature',
        type: 'Number',
        metadata: {
          ga: {
            value: 'thermostatTemperatureSetpoint'
          }
        },
        state: '10'
      }, {
        type: 'Number',
        metadata: {
          ga: {
            value: 'thermostatMode'
          }
        },
        state: 'heat'
      }, {
        type: 'Number',
        metadata: {
          ga: {
            value: 'thermostatHumidityAmbient'
          }
        },
        state: '50'
      }]
    };

    const getItemMock = jest.fn();
    const sendCommandMock = jest.fn();
    getItemMock.mockReturnValue(Promise.resolve(item));
    sendCommandMock.mockReturnValue(Promise.resolve());

    const apiHandler = {
      getItem: getItemMock,
      sendCommand: sendCommandMock
    };

    const commands = [{
      "devices": [{
        "customData": {
          "thermostatTemperatureSetpoint": "MyTargetTemperature"
        },
        "id": "MyThermostat"
      }],
      "execution": [{
        "command": "action.devices.commands.ThermostatTemperatureSetpoint",
        "params": {
          "thermostatTemperatureSetpoint": 25
        }
      }]
    }];

    const payload = await new OpenHAB(apiHandler).handleExecute(commands);

    expect(getItemMock).toHaveBeenCalledTimes(1);
    expect(sendCommandMock).toBeCalledWith('MyTargetTemperature', '77');
    expect(payload).toStrictEqual({
      "commands": [{
        "ids": [
          "MyThermostat"
        ],
        "states": {
          "online": true,
          "thermostatHumidityAmbient": 50,
          "thermostatMode": "heat",
          "thermostatTemperatureAmbient": -12.2,
          "thermostatTemperatureSetpoint": 25
        },
        "status": "SUCCESS"
      }]
    });
  });

  test('ThermostatTemperatureSetRange', async () => {
    const item1 =
    {
      "state": "NULL",
      "type": "Group",
      "name": "MyThermostat",
      "label": "Thermostat",
      "metadata": {
        "ga": {
          "value": "Thermostat"
        }
      },
      "members": [{
        type: 'Number',
        metadata: {
          ga: {
            value: 'thermostatTemperatureAmbient'
          }
        },
        state: '10'
      }, {
        type: 'Number',
        metadata: {
          ga: {
            value: 'thermostatTemperatureSetpoint'
          }
        },
        state: '10'
      }, {
        name: 'MyTargetTemperatureHigh',
        type: 'Number',
        metadata: {
          ga: {
            value: 'thermostatTemperatureSetpointHigh'
          }
        },
        state: '13'
      }, {
        name: 'MyTargetTemperatureLow',
        type: 'Number',
        metadata: {
          ga: {
            value: 'thermostatTemperatureSetpointLow'
          }
        },
        state: '7'
      }, {
        type: 'Number',
        metadata: {
          ga: {
            value: 'thermostatMode'
          }
        },
        state: 'heatcool'
      }, {
        type: 'Number',
        metadata: {
          ga: {
            value: 'thermostatHumidityAmbient'
          }
        },
        state: '50'
      }]
    };

    const item2 =
    {
      "state": "NULL",
      "type": "Group",
      "name": "MyThermostat",
      "label": "Thermostat",
      "metadata": {
        "ga": {
          "value": "Thermostat"
        }
      },
      "members": [{
        type: 'Number',
        metadata: {
          ga: {
            value: 'thermostatTemperatureAmbient'
          }
        },
        state: '10'
      }, {
        type: 'Number',
        metadata: {
          ga: {
            value: 'thermostatTemperatureSetpoint'
          }
        },
        state: '10'
      }, {
        name: 'MyTargetTemperatureHigh',
        type: 'Number',
        metadata: {
          ga: {
            value: 'thermostatTemperatureSetpointHigh'
          }
        },
        state: '25'
      }, {
        name: 'MyTargetTemperatureLow',
        type: 'Number',
        metadata: {
          ga: {
            value: 'thermostatTemperatureSetpointLow'
          }
        },
        state: '7'
      }, {
        type: 'Number',
        metadata: {
          ga: {
            value: 'thermostatMode'
          }
        },
        state: 'heatcool'
      }, {
        type: 'Number',
        metadata: {
          ga: {
            value: 'thermostatHumidityAmbient'
          }
        },
        state: '50'
      }]
    };

    const getItemMock = jest.fn();
    const sendCommandMock = jest.fn();
    getItemMock.mockReturnValueOnce(Promise.resolve(item1))
      .mockReturnValueOnce(Promise.resolve(item2));
    sendCommandMock.mockReturnValue(Promise.resolve());

    const apiHandler = {
      getItem: getItemMock,
      sendCommand: sendCommandMock
    };

    const commands = [{
      "devices": [{
        "customData": {
          "thermostatTemperatureSetpointHigh": "MyTargetTemperatureHigh",
          "thermostatTemperatureSetpointLow": "MyTargetTemperatureLow"
        },
        "id": "MyThermostat"
      }],
      "execution": [{
        "command": "action.devices.commands.ThermostatTemperatureSetRange",
        "params": {
          "thermostatTemperatureSetpointHigh": 25,
          "thermostatTemperatureSetpointLow": 15
        }
      }]
    }];

    const payload = await new OpenHAB(apiHandler).handleExecute(commands);

    expect(getItemMock).toHaveBeenCalledTimes(2);
    expect(sendCommandMock).toBeCalledWith('MyTargetTemperatureHigh', '25');
    expect(sendCommandMock).toBeCalledWith('MyTargetTemperatureLow', '15');
    expect(payload).toStrictEqual({
      "commands": [{
        "ids": [
          "MyThermostat"
        ],
        "states": {
          "online": true,
          "thermostatHumidityAmbient": 50,
          "thermostatMode": "heatcool",
          "thermostatTemperatureAmbient": 10,
          "thermostatTemperatureSetpoint": 10,
          "thermostatTemperatureSetpointHigh": 25,
          "thermostatTemperatureSetpointLow": 15
        },
        "status": "SUCCESS"
      }]
    });
  });

  test('ThermostatSetMode invalid', async () => {
    const item =
    {
      "state": "NULL",
      "type": "Group",
      "name": "MyThermostat",
      "label": "Thermostat",
      "metadata": {
        "ga": {
          "value": "Thermostat",
          "config": {
            "modes": "on=3,heat=5"
          }
        }
      },
      "members": [{
        type: 'Number',
        metadata: {
          ga: {
            value: 'thermostatTemperatureAmbient'
          }
        },
        state: '20'
      }, {
        name: 'MyTargetTemperature',
        type: 'Number',
        metadata: {
          ga: {
            value: 'thermostatTemperatureSetpoint'
          }
        },
        state: '10'
      }, {
        type: 'Number',
        metadata: {
          ga: {
            value: 'thermostatMode'
          }
        },
        state: '3'
      }, {
        type: 'Number',
        metadata: {
          ga: {
            value: 'thermostatHumidityAmbient'
          }
        },
        state: '50'
      }]
    };

    const getItemMock = jest.fn();
    const sendCommandMock = jest.fn();
    getItemMock.mockReturnValue(Promise.resolve(item));
    sendCommandMock.mockReturnValue(Promise.resolve());

    const apiHandler = {
      getItem: getItemMock,
      sendCommand: sendCommandMock
    };

    const commands = [{
      "devices": [{
        "customData": {
          "thermostatMode": "MyMode"
        },
        "id": "MyThermostat"
      }],
      "execution": [{
        "command": "action.devices.commands.ThermostatSetMode",
        "params": {
          "thermostatMode": "off"
        }
      }]
    }];

    const payload = await new OpenHAB(apiHandler).handleExecute(commands);

    expect(getItemMock).toHaveBeenCalledTimes(1);
    expect(sendCommandMock).toHaveBeenCalledTimes(0);
    expect(payload).toStrictEqual({
      "commands": [{
        "ids": [
          "MyThermostat"
        ],
        "errorCode": "notSupported",
        "status": "ERROR"
      }]
    });
  });

  test('ThermostatSetMode', async () => {
    const item =
    {
      "state": "NULL",
      "type": "Group",
      "name": "MyThermostat",
      "label": "Thermostat",
      "metadata": {
        "ga": {
          "value": "Thermostat",
          "config": {
            "modes": "on=1,off=5"
          }
        }
      },
      "members": [{
        type: 'Number',
        metadata: {
          ga: {
            value: 'thermostatTemperatureAmbient'
          }
        },
        state: '20'
      }, {
        name: 'MyTargetTemperature',
        type: 'Number',
        metadata: {
          ga: {
            value: 'thermostatTemperatureSetpoint'
          }
        },
        state: '10'
      }, {
        type: 'Number',
        metadata: {
          ga: {
            value: 'thermostatMode'
          }
        },
        state: '1'
      }, {
        type: 'Number',
        metadata: {
          ga: {
            value: 'thermostatHumidityAmbient'
          }
        },
        state: '50'
      }]
    };

    const getItemMock = jest.fn();
    const sendCommandMock = jest.fn();
    getItemMock.mockReturnValue(Promise.resolve(item));
    sendCommandMock.mockReturnValue(Promise.resolve());

    const apiHandler = {
      getItem: getItemMock,
      sendCommand: sendCommandMock
    };

    const commands = [{
      "devices": [{
        "customData": {
          "thermostatMode": "MyMode"
        },
        "id": "MyThermostat"
      }],
      "execution": [{
        "command": "action.devices.commands.ThermostatSetMode",
        "params": {
          "thermostatMode": "off"
        }
      }]
    }];

    const payload = await new OpenHAB(apiHandler).handleExecute(commands);

    expect(getItemMock).toHaveBeenCalledTimes(1);
    expect(sendCommandMock).toBeCalledWith('MyMode', '5');
    expect(payload).toStrictEqual({
      "commands": [{
        "ids": [
          "MyThermostat"
        ],
        "states": {
          "online": true,
          "thermostatHumidityAmbient": 50,
          "thermostatMode": "off",
          "thermostatTemperatureAmbient": 20,
          "thermostatTemperatureSetpoint": 10
        },
        "status": "SUCCESS"
      }]
    });
  });

  test('Lock with required acknowledge', async () => {
    const item =
    {
      "state": "OFF",
      "type": "Switch",
      "name": "MyLock",
      "label": "My Lock",
      "metadata": {
        "ga": {
          "value": "Lock",
          "config": {
            "tfaAck": true
          }
        }
      },
      "tags": []
    };

    const getItemMock = jest.fn();
    const sendCommandMock = jest.fn();
    getItemMock.mockReturnValue(Promise.resolve(item));
    sendCommandMock.mockReturnValue(Promise.resolve());

    const apiHandler = {
      getItem: getItemMock,
      sendCommand: sendCommandMock
    };

    const commands = [{
      "devices": [{
        "customData": {
          "tfaAck": true
        },
        "id": "MyLock"
      }],
      "execution": [{
        "command": "action.devices.commands.LockUnlock",
        "params": {
          lock: true
        }
      }]
    }];

    const payload = await new OpenHAB(apiHandler).handleExecute(commands);

    expect(getItemMock).toHaveBeenCalledTimes(1);
    expect(payload).toStrictEqual({
      "commands": [{
        "ids": [
          "MyLock"
        ],
        "states": {
          "online": true,
          "isLocked": true
        },
        "status": "ERROR",
        "errorCode": "challengeNeeded",
        "challengeNeeded": {
          "type": "ackNeeded",
        }
      }]
    });
  });

  test('Lock with acknowledged challenge', async () => {
    const getItemMock = jest.fn();
    const sendCommandMock = jest.fn();
    getItemMock.mockReturnValue(Promise.resolve());
    sendCommandMock.mockReturnValue(Promise.resolve());

    const apiHandler = {
      getItem: getItemMock,
      sendCommand: sendCommandMock
    };

    const commands = [{
      "devices": [{
        "customData": {
          "tfaAck": true
        },
        "id": "MyLock"
      }],
      "execution": [{
        "command": "action.devices.commands.LockUnlock",
        "params": {
          lock: true
        },
        "challenge": {
          "ack": true
        }
      }]
    }];

    const payload = await new OpenHAB(apiHandler).handleExecute(commands);

    expect(getItemMock).toHaveBeenCalledTimes(0);
    expect(payload).toStrictEqual({
      "commands": [{
        "ids": [
          "MyLock"
        ],
        "states": {
          "online": true,
          "isLocked": true
        },
        "status": "SUCCESS"
      }]
    });
  });

  test('BrightnessAbsolute with required acknowledge', async () => {
    const getItemMock = jest.fn();
    const sendCommandMock = jest.fn();
    getItemMock.mockReturnValue(Promise.resolve());
    sendCommandMock.mockReturnValue(Promise.resolve());

    const apiHandler = {
      getItem: getItemMock,
      sendCommand: sendCommandMock
    };

    const commands = [{
      "devices": [{
        "customData": {
          "tfaAck": true
        },
        "id": "MyLight"
      }],
      "execution": [{
        "command": "action.devices.commands.BrightnessAbsolute",
        "params": {
          "brightness": 30
        }
      }]
    }];

    const payload = await new OpenHAB(apiHandler).handleExecute(commands);

    expect(getItemMock).toHaveBeenCalledTimes(0);
    expect(payload).toStrictEqual({
      "commands": [{
        "ids": [
          "MyLight"
        ],
        "states": {
          "online": true,
          "brightness": 30
        },
        "status": "ERROR",
        "errorCode": "challengeNeeded",
        "challengeNeeded": {
          "type": "ackNeeded",
        }
      }]
    });
  });

  test('Arm with required pin', async () => {
    const getItemMock = jest.fn();
    const sendCommandMock = jest.fn();
    getItemMock.mockReturnValue(Promise.resolve());
    sendCommandMock.mockReturnValue(Promise.resolve());

    const apiHandler = {
      getItem: getItemMock,
      sendCommand: sendCommandMock
    };

    const commands = [{
      "devices": [{
        "customData": {
          "tfaPin": "1234"
        },
        "id": "MyAlarm"
      }],
      "execution": [{
        "command": "action.devices.commands.ArmDisarm",
        "params": {
          arm: true
        }
      }]
    }];

    const payload = await new OpenHAB(apiHandler).handleExecute(commands);

    expect(getItemMock).toHaveBeenCalledTimes(0);
    expect(payload).toStrictEqual({
      "commands": [{
        "ids": [
          "MyAlarm"
        ],
        "status": "ERROR",
        "errorCode": "challengeNeeded",
        "challengeNeeded": {
          "type": "pinNeeded",
        }
      }]
    });
  });

  test('Arm with wrong pin', async () => {
    const getItemMock = jest.fn();
    const sendCommandMock = jest.fn();
    getItemMock.mockReturnValue(Promise.resolve());
    sendCommandMock.mockReturnValue(Promise.resolve());

    const apiHandler = {
      getItem: getItemMock,
      sendCommand: sendCommandMock
    };

    const commands = [{
      "devices": [{
        "customData": {
          "tfaPin": "1234"
        },
        "id": "MyAlarm"
      }],
      "execution": [{
        "command": "action.devices.commands.ArmDisarm",
        "params": {
          "arm": true
        },
        "challenge": {
          "pin": "3456"
        }
      }]
    }];

    const payload = await new OpenHAB(apiHandler).handleExecute(commands);

    expect(getItemMock).toHaveBeenCalledTimes(0);
    expect(payload).toStrictEqual({
      "commands": [{
        "ids": [
          "MyAlarm"
        ],
        "status": "ERROR",
        "errorCode": "challengeNeeded",
        "challengeNeeded": {
          "type": "challengeFailedPinNeeded"
        }
      }]
    });
  });

  test('Arm with correct pin', async () => {
    const getItemMock = jest.fn();
    const sendCommandMock = jest.fn();
    getItemMock.mockReturnValue(Promise.resolve());
    sendCommandMock.mockReturnValue(Promise.resolve());

    const apiHandler = {
      getItem: getItemMock,
      sendCommand: sendCommandMock
    };

    const commands = [{
      "devices": [{
        "customData": {
          "tfaPin": "1234"
        },
        "id": "MyAlarm"
      }],
      "execution": [{
        "command": "action.devices.commands.ArmDisarm",
        "params": {
          "arm": true
        },
        "challenge": {
          "pin": "1234"
        }
      }]
    }];

    const payload = await new OpenHAB(apiHandler).handleExecute(commands);

    expect(getItemMock).toHaveBeenCalledTimes(0);
    expect(payload).toStrictEqual({
      "commands": [{
        "ids": [
          "MyAlarm"
        ],
        "states": {
          "online": true,
          "isArmed": true
        },
        "status": "SUCCESS"
      }]
    });
  });
});
