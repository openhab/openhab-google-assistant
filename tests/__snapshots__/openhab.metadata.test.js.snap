// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SYNC with Metadata Fan Device 1`] = `
Object {
  "devices": Array [
    Object {
      "attributes": Object {
        "availableFanSpeeds": Object {
          "ordered": true,
          "speeds": Array [
            Object {
              "speed_name": "0",
              "speed_values": Array [
                Object {
                  "lang": "en",
                  "speed_synonym": Array [
                    "null",
                    "off",
                  ],
                },
              ],
            },
            Object {
              "speed_name": "50",
              "speed_values": Array [
                Object {
                  "lang": "en",
                  "speed_synonym": Array [
                    "slow",
                  ],
                },
              ],
            },
            Object {
              "speed_name": "100",
              "speed_values": Array [
                Object {
                  "lang": "en",
                  "speed_synonym": Array [
                    "full",
                    "fast",
                  ],
                },
              ],
            },
          ],
        },
        "reversible": false,
      },
      "customData": Object {
        "deviceType": "action.devices.types.FAN",
        "itemType": "Dimmer",
        "tfaAck": undefined,
        "tfaPin": undefined,
      },
      "deviceInfo": Object {
        "hwVersion": "2.5.0",
        "manufacturer": "openHAB",
        "model": "Dimmer",
        "swVersion": "2.5.0",
      },
      "id": "MyFan",
      "name": Object {
        "defaultNames": Array [
          "My Fan",
        ],
        "name": "My Fan",
        "nicknames": Array [
          "My Fan",
        ],
      },
      "roomHint": undefined,
      "structureHint": undefined,
      "traits": Array [
        "action.devices.traits.OnOff",
        "action.devices.traits.FanSpeed",
      ],
      "type": "action.devices.types.FAN",
      "willReportState": false,
    },
  ],
}
`;

exports[`Test SYNC with Metadata Light Devices 1`] = `
Object {
  "devices": Array [
    Object {
      "attributes": Object {},
      "customData": Object {
        "deviceType": "action.devices.types.LIGHT",
        "itemType": "Switch",
        "tfaAck": undefined,
        "tfaPin": undefined,
      },
      "deviceInfo": Object {
        "hwVersion": "2.5.0",
        "manufacturer": "openHAB",
        "model": "Switch",
        "swVersion": "2.5.0",
      },
      "id": "MySwitch",
      "name": Object {
        "defaultNames": Array [
          "Light Switch",
        ],
        "name": "Light Switch",
        "nicknames": Array [
          "Light Switch",
          "Testlight",
          "Cool Light",
        ],
      },
      "roomHint": undefined,
      "structureHint": undefined,
      "traits": Array [
        "action.devices.traits.OnOff",
      ],
      "type": "action.devices.types.LIGHT",
      "willReportState": false,
    },
    Object {
      "attributes": Object {},
      "customData": Object {
        "deviceType": "action.devices.types.LIGHT",
        "itemType": "Dimmer",
        "tfaAck": undefined,
        "tfaPin": undefined,
      },
      "deviceInfo": Object {
        "hwVersion": "2.5.0",
        "manufacturer": "openHAB",
        "model": "Dimmer",
        "swVersion": "2.5.0",
      },
      "id": "MyDimmer",
      "name": Object {
        "defaultNames": Array [
          "DimmLight",
        ],
        "name": "DimmLight",
        "nicknames": Array [
          "DimmLight",
        ],
      },
      "roomHint": undefined,
      "structureHint": undefined,
      "traits": Array [
        "action.devices.traits.OnOff",
        "action.devices.traits.Brightness",
      ],
      "type": "action.devices.types.LIGHT",
      "willReportState": false,
    },
    Object {
      "attributes": Object {
        "colorModel": "hsv",
      },
      "customData": Object {
        "deviceType": "action.devices.types.LIGHT",
        "itemType": "Color",
        "tfaAck": undefined,
        "tfaPin": undefined,
      },
      "deviceInfo": Object {
        "hwVersion": "2.5.0",
        "manufacturer": "openHAB",
        "model": "Color",
        "swVersion": "2.5.0",
      },
      "id": "MyLight",
      "name": Object {
        "defaultNames": Array [
          "ColorLight",
        ],
        "name": "ColorLight",
        "nicknames": Array [
          "ColorLight",
        ],
      },
      "roomHint": undefined,
      "structureHint": undefined,
      "traits": Array [
        "action.devices.traits.OnOff",
        "action.devices.traits.Brightness",
        "action.devices.traits.ColorSetting",
      ],
      "type": "action.devices.types.LIGHT",
      "willReportState": false,
    },
    Object {
      "attributes": Object {},
      "customData": Object {
        "deviceType": "action.devices.types.LIGHT",
        "itemType": "Switch",
        "tfaAck": undefined,
        "tfaPin": undefined,
      },
      "deviceInfo": Object {
        "hwVersion": "2.5.0",
        "manufacturer": "openHAB",
        "model": "Group",
        "swVersion": "2.5.0",
      },
      "id": "MyLightGroup",
      "name": Object {
        "defaultNames": Array [
          "GroupLight",
        ],
        "name": "GroupLight",
        "nicknames": Array [
          "GroupLight",
        ],
      },
      "roomHint": undefined,
      "structureHint": undefined,
      "traits": Array [
        "action.devices.traits.OnOff",
      ],
      "type": "action.devices.types.LIGHT",
      "willReportState": false,
    },
    Object {
      "attributes": Object {},
      "customData": Object {
        "deviceType": "action.devices.types.LIGHT",
        "itemType": "Dimmer",
        "tfaAck": undefined,
        "tfaPin": undefined,
      },
      "deviceInfo": Object {
        "hwVersion": "2.5.0",
        "manufacturer": "openHAB",
        "model": "Group",
        "swVersion": "2.5.0",
      },
      "id": "MyDimmerGroup",
      "name": Object {
        "defaultNames": Array [
          "GroupDimmer",
        ],
        "name": "GroupDimmer",
        "nicknames": Array [
          "GroupDimmer",
        ],
      },
      "roomHint": undefined,
      "structureHint": undefined,
      "traits": Array [
        "action.devices.traits.OnOff",
        "action.devices.traits.Brightness",
      ],
      "type": "action.devices.types.LIGHT",
      "willReportState": false,
    },
    Object {
      "attributes": Object {
        "colorModel": "hsv",
      },
      "customData": Object {
        "deviceType": "action.devices.types.LIGHT",
        "itemType": "Color",
        "tfaAck": undefined,
        "tfaPin": undefined,
      },
      "deviceInfo": Object {
        "hwVersion": "2.5.0",
        "manufacturer": "openHAB",
        "model": "Group",
        "swVersion": "2.5.0",
      },
      "id": "MyColorGroup",
      "name": Object {
        "defaultNames": Array [
          "GroupColor",
        ],
        "name": "GroupColor",
        "nicknames": Array [
          "GroupColor",
        ],
      },
      "roomHint": undefined,
      "structureHint": undefined,
      "traits": Array [
        "action.devices.traits.OnOff",
        "action.devices.traits.Brightness",
        "action.devices.traits.ColorSetting",
      ],
      "type": "action.devices.types.LIGHT",
      "willReportState": false,
    },
  ],
}
`;

exports[`Test SYNC with Metadata Scene Device 1`] = `
Object {
  "devices": Array [
    Object {
      "attributes": Object {
        "sceneReversible": true,
      },
      "customData": Object {
        "deviceType": "action.devices.types.SCENE",
        "itemType": "Switch",
        "tfaAck": undefined,
        "tfaPin": undefined,
      },
      "deviceInfo": Object {
        "hwVersion": "2.5.0",
        "manufacturer": "openHAB",
        "model": "Switch",
        "swVersion": "2.5.0",
      },
      "id": "MyScene",
      "name": Object {
        "defaultNames": Array [
          "My Scene",
        ],
        "name": "My Scene",
        "nicknames": Array [
          "My Scene",
        ],
      },
      "roomHint": undefined,
      "structureHint": undefined,
      "traits": Array [
        "action.devices.traits.Scene",
      ],
      "type": "action.devices.types.SCENE",
      "willReportState": false,
    },
  ],
}
`;
